src/fsharp/CompileOps.fs

```diff
@@ -1643,6 +1643,7 @@ let GetFsiLibraryName () = "FSharp.Compiler.Interactive.Settings"
 // .NET Core references
 let DefaultBasicReferencesForOutOfProjectSources = 
     [   yield Path.Combine(Path.GetDirectoryName(typeof<System.Object>.Assembly.Location),"mscorlib.dll"); // mscorlib
+        yield typeof<System.Object>.Assembly.Location; // System.Private.CoreLib
```

Comment:
Yes, we were not testing this particular scenario on netcore. I added a test for it, here is the error if the reference to `System.Private.CoreLib` is not there:
```
Project35b error: <<<The type 'Object' is required here and is unavailable. You must add a reference to assembly 'System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e'.>>>
Project35b error: <<<A reference to the type 'System.Object' in assembly 'mscorlib' was found, but the type could not be found in that assembly>>>
```